docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,131 @@
+
+While this process has nominally worked for several releases of EdgeX, there has always been some issues with this extra initialization process (config-seed), not least of which are:
+- race conditions on the part of the services, as they bootstrap, coming up before the config-seed completes its deposit of configuration into Consul
+- how to deal with "overrides" such as environmental variable provided configuration overrides. As the override is often specific to a service but has to be in place for config-seed in order to take effect.
+- need for an additional service that is only there for init and then dies (confusing to users)
+
+NOTE - for historical purposes, it should be noted that config-seed only writes configuration into the configuration/registry service (Consul) once on the first start of EdgeX.  On subsequent starts of EdgeX, config-seed checks to see if it has already populated the configuration/registry service and will not rewrite configuration again (unless the --overwrite flag is used).
+
+The design/architectural proposal, therefore, is:
+- removal of the config-seed service (removing cmd/config-seed from the edgex-go repository)
+- have each EdgeX micro service "self seed" - that is seed Consul with their own required configuration on bootstrap of the service.  Details of that bootstrapping process are below.
+
+### Command Line Options
+All EdgeX services support a common set of command-line options, some combination of which are required on startup for a service to interact with the rest of EdgeX. Command line options are not set by any configuration.  Command line options include:
+
+- --configProvider or -cp (the configuration provider location URL)
+- --overwrite or -o (overwrite the configuration in the configuration provider)
+- --file or -f (the configuration filename - configuration.toml is used by default if the configuration filename is not provided)
+- --profile or -p (the name of a sub directory in which the profile-specific configuration file is found.  This is either the sub directory of ./res directory by default or the sub directory of the confdir directory when specified )
+- --confdir or -c (the directory where the configuration file is found - ./res is used by default if the confdir is not specified, where "." is the convention on Linux/Unix/MacOS which means current directory) 
+- --registry or -r (string indicating use of the registry)
+
+The distinction of command line options versus configuration will be important later in this ADR.
+
+Two command line options (-o for overwrite and -r for registry) are not overridable by environmental variables.
+
+NOTES:
+- For backwards compatibility, there is an edgex_registry environment variable that will override the  --configProvider / -cp and --registry / -r command line options.
```

Comment:
This doesn't fully make sense and is actually described in better detail in the Backwards Compatibility section. Suggest you drop it here.

Reply:
I don't see why this doesn't make sense, but fine with just having what is on the `Backwards Compatibility` section.

Reply:
Will remove to resolve.

Reply:
thanks